tools/extensions/validate_extensions.py

```diff
@@ -0,0 +1,238 @@
+]
+
+EXTENSION_STATUS_VALUES = [
+    # This extension is stable and is expected to be production usable.
+    "stable",
+    # This extension is functional but has not had substantial production burn
+    # time, use only with this caveat.
+    "alpha",
+    # This extension is work-in-progress. Functionality is incomplete and it is
+    # not intended for production use.
+    "wip",
+]
+
+BUILDOZER_PATH = os.path.abspath(
+    "external/com_github_bazelbuild_buildtools/buildozer/buildozer_/buildozer")
+
+ENVOY_SRCDIR = os.getenv('ENVOY_SRCDIR', '/source')
+
+if not os.path.exists(ENVOY_SRCDIR):
+    raise SystemExit(
+        "Envoy source must either be located at /source, or ENVOY_SRCDIR env var must be set")
+
+# source/extensions/extensions_build_config.bzl must have a .bzl suffix for Starlark
+# import, so we are forced to do this workaround.
+_extensions_build_config_spec = spec_from_loader(
+    'extensions_build_config',
+    SourceFileLoader(
+        'extensions_build_config',
+        os.path.join(ENVOY_SRCDIR, 'source/extensions/extensions_build_config.bzl')))
+extensions_build_config = module_from_spec(_extensions_build_config_spec)
+_extensions_build_config_spec.loader.exec_module(extensions_build_config)
+
+
+class ExtensionDbError(Exception):
+    pass
+
+
+def is_missing(value):
+    return value == '(missing)'
+
+
+def num_read_filters_fuzzed():
+    data = pathlib.Path(
+        os.path.join(
+            ENVOY_SRCDIR,
+            'test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc')).read_text()
+    # Hack-ish! We only search the first 50 lines to capture the filters in filterNames().
+    return len(re.findall('NetworkFilterNames::get()', ''.join(data.splitlines()[:50])))
+
+
+def num_robust_to_downstream_network_filters(db):
+    # Count number of network filters robust to untrusted downstreams.
+    return len([
+        ext for ext, data in db.items()
+        if 'network' in ext and data['security_posture'] == 'robust_to_untrusted_downstream'
+    ])
+
+
+def get_extension_metadata(target):
+    if not BUILDOZER_PATH:
+        raise ExtensionDbError('Buildozer not found!')
+    r = subprocess.run(
+        [BUILDOZER_PATH, '-stdout', 'print security_posture status undocumented category', target],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+    rout = r.stdout.decode('utf-8').strip().split(' ')
+    security_posture, status, undocumented = rout[:3]
+    categories = ' '.join(rout[3:])
+    # evaluate tuples/lists
+    # wrap strings in a list
+    categories = list(
+        ast.literal_eval(categories) if ('[' in categories or '(' in categories) else [categories])
+    return {
+        'security_posture': security_posture,
+        'undocumented': False if is_missing(undocumented) else bool(undocumented),
+        'status': 'stable' if is_missing(status) else status,
+        'categories': categories,
+    }
+
+
+def compare_old_and_new(old_db, new_db):
```

Comment:
Is this just to validate we're doing the right thing as scaffolding? Should it be deleted before merging?

Reply:
yes and no - its to ensure that transitioning from the old macro way of setting the metadata and the new yaml way nothing is lost

Reply:
...and for that reason i think its important to keep it until the macro way has been removed completely

Reply:
If you want to preserve this over two PRs, then can you mark everything with a comment that will be gone in the next PR, so we don't have to review it with scrutiny?

Reply:
ive added TODOs to all of the soon-to-be-removed code